src/librustc/middle/liveness.rs

```diff
@@ -1490,9 +1492,17 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
 
         match fn_ret {
             ty::FnConverging(t_ret)
-                if self.live_on_entry(entry_ln, self.s.no_ret_var).is_some() => {
+                    if self.live_on_entry(entry_ln, self.s.no_ret_var).is_some() => {
 
-                if t_ret.is_nil() {
+                let infcx = infer::new_infer_ctxt(&self.ir.tcx, &self.ir.tcx.tables, None);
+                let mut selcx = traits::SelectionContext::new(&infcx);
+                let cause = traits::ObligationCause::dummy();
+
+                let norm = traits::normalize(&mut selcx,
+                                             cause,
+                                             &t_ret);
+
```

Comment:
What should I do with possible errors? I'm assuming they would already be reported elsewhere, so should I just drop them and assume the result is (or is not) `()`? Or can errors not occur here, since liveness runs so late in the pipeline?